### PR TITLE
Improve Automation Coverage for Critical path of iOS

### PR DESCRIPTION
**Pull request by bilalawan321**
[LEARNER-8440](https://openedx.atlassian.net/browse/LEARNER-8440)

The following scenarios is implemented:

Update strings for login and new landing screen
Add test cases for reset password and wrong credentials in the login screen
Verify that, if the user fills all data except "Full Name" and tap on the 'Create my account' button,
an alert pop-up should appear along with the error message.
Verify that, if the user fills all data except "User Name" and tap on the 'Create my account' button,
an alert pop-up should appear along with the error message.
Verify that, if the user fills all data except "Password" and tap on the 'Create my account' button,
an alert pop-up should appear along with the error message.
Verify that, if user fill all data but with wrong email format and tap on 'Create my account' button,
email format error message should appear.
Verify that, if user fill all data but with wrong email format and tap on 'Create my account' button,
email format error message should appear.
Verify on tapping Programs will load Programs contents in its tab
Verify on tapping Programs will load Programs contents in its tab in landscape mode
Remove coding=utf-8 from android pages and tests files

### CODE REVIEW

File: tests/ios/pages/ios_register.py [hunk 1]
```diff
@@ -718,5 +722,5 @@ def select_country(self):
         """
 
         self.get_country_spinner().click()
-        self.global_contents.wait_for_element_visibility(self.driver, ios_elements.select_country)
-        self.global_contents.wait_and_get_element(self.driver, ios_elements.select_country).click()
+        # self.global_contents.wait_for_element_visibility(self.driver, ios_elements.select_country)
```
Comment (jawad-khan) [inline tests/ios/pages/ios_register.py:725]: is this code commented intentionally?
if yes please add reason too.

File: tests/ios/tests/test_ios_login.py [hunk 2]
```diff
@@ -181,9 +199,9 @@ def test_landscape_smoke(self, set_capabilities, setup_logging):
 
         assert ios_login_page.back_and_forth_new_landing()
         # global_contents.scroll_from_element(set_capabilities, ios_login_page.get_forgot_password_textview())
-        assert ios_login_page.load_eula_screen().text == strings.LOGIN
-        assert ios_login_page.load_terms_screen().text == strings.LOGIN
-        assert ios_login_page.load_privacy_screen().text == strings.LOGIN
+        # assert ios_login_page.load_eula_screen().text == strings.LOGIN
```
Comment (jawad-khan) [inline tests/ios/tests/test_ios_login.py:202]: is this code commented intentionally?
if yes please add reason too.

**Review by jawad-khan — APPROVED** [review]

File: tests/ios/pages/ios_register.py (hunk 1 above)
Comment (bilalawan321) [reply tests/ios/pages/ios_register.py]: fixed

File: tests/ios/tests/test_ios_login.py (hunk 2 above)
Comment (bilalawan321) [reply tests/ios/tests/test_ios_login.py:203]: fixed

**Pull request merged**